File: Repaso 6-10-24/2.js
function binarySearch(n, arr) {
  if (arr.length == 0) return false;

  const m = Math.floor(arr.length / 2);
  console.log("m", m);

  if (n == arr[m]) return true;

  return n > arr[m]
    ? binarySearch(n, arr.slice(m + 1))
    : binarySearch(n, arr.slice(0, m));
}

const arr = [1, 2, 3, 4, 5, 6];
console.log(binarySearch(2, arr));

File: Repaso 6-10-24/2.py
def binarySearch(n, arr):
    if len(arr) == 0:
        return False

    m = len(arr) // 2
    print("len", len(arr), "m", m)

    if n == arr[m]:
        return True

    if n > arr[m]:
        return binarySearch(n, arr[m + 1 :])
    return binarySearch(n, arr[:m])


arr = [1, 2, 3, 4, 5, 6]
print(binarySearch(5, arr))
